src/deepsparse/transformers/eval_downstream.py

```diff
@@ -0,0 +1,283 @@
+        mnli_metrics.add_batch(
+            predictions=[int(pred[0]["label"].split("_")[-1])],
+            references=[sample["label"]],
+        )
+
+        if args.max_samples and idx > args.max_samples:
+            break
+
+    for idx, sample in enumerate(tqdm(mnli_mismatched)):
+        pred = text_classify(sample["premise"], sample["hypothesis"])
+        mnli_metrics.add_batch(
+            predictions=[int(pred[0]["label"].split("_")[-1])],
+            references=[sample["label"]],
+        )
+
+        if args.max_samples and idx > args.max_samples:
+            break
+
+    print(f"\nMNLI eval results: {mnli_metrics.compute()}")
+
+
+def qqp_eval(args):
+    # load qqp validation dataset and eval tool
+    qqp = load_dataset("glue", "qqp")["validation"]
+    qqp_metrics = load_metric("glue", "qqp")
+
+    # load pipeline
+    text_classify = pipeline(
+        task="text-classification",
+        model_path=args.onnx_filepath,
+        engine_type=args.engine,
+        num_cores=args.num_cores,
+        max_length=args.max_sequence_length,
+    )
+    print(f"Engine info: {text_classify.model}")
+
+    for idx, sample in enumerate(tqdm(qqp)):
+        pred = text_classify(sample["question1"], sample["question2"])
+
+        qqp_metrics.add_batch(
+            predictions=[int(pred[0]["label"].split("_")[-1])],
+            references=[sample["label"]],
+        )
+
+        if args.max_samples and idx > args.max_samples:
+            break
+
+    print(f"\nQQP eval results: {qqp_metrics.compute()}")
+
+
+def sst2_eval(args):
+    # load sst2 validation dataset and eval tool
+    sst2 = load_dataset("glue", "sst2")["validation"]
+    sst2_metrics = load_metric("glue", "sst2")
+
+    # load pipeline
+    text_classify = pipeline(
+        task="text-classification",
+        model_path=args.onnx_filepath,
+        engine_type=args.engine,
+        num_cores=args.num_cores,
+        max_length=args.max_sequence_length,
+    )
+    print(f"Engine info: {text_classify.model}")
+
+    for idx, sample in enumerate(tqdm(sst2)):
+        pred = text_classify(
+            sample["sentence"],
+        )
+
+        sst2_metrics.add_batch(
+            predictions=[int(pred[0]["label"].split("_")[-1])],
+            references=[sample["label"]],
+        )
+
+        if args.max_samples and idx > args.max_samples:
+            break
+
+    print(f"\nSST-2 eval results: {sst2_metrics.compute()}")
+
+
+def main():
+    args = parse_args()
+
+    args.dataset = args.dataset.lower()
+    if args.dataset == SQUAD_KEY:
+        squad_eval(args)
+    elif args.dataset == MNLI_KEY:
+        mnli_eval(args)
+    elif args.dataset == QQP_KEY:
+        qqp_eval(args)
+    elif args.dataset == SST2_KEY:
+        sst2_eval(args)
+    else:
+        raise KeyError(
+            f"Unknown downstream dataset {args.dataset}, "
+            f"available datasets are {SUPPORTED_DATASETS}"
+        )
```

Comment:
Can be removed, `argparse` more specifically line 90 takes care of this for us if invalid dataset provided,

```
eval_downstream.py: error: argument -d/--dataset: invalid choice: 's' (choose from 'squad', 'mnli', 'qqp', 'sst2')
```

```suggestion
```